File: VB/XPO/ViewSettings/ViewSettings.Module/BusinessObjects/SettingsStore.vb
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp
Imports System.ComponentModel
Imports DevExpress.Xpo

Namespace ViewSettingsSolution.Module.BusinessObjects

    Public Class SettingsStore
        Inherits BaseObject
        Implements IObjectSpaceLink

        Private xmlField As String

        Private nameField As String

        Private ownerIdField As String

        Private viewIdField As String

        Private isSharedField As Boolean

        Private objectSpaceField As IObjectSpace

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        <Browsable(False)>
        <Size(SizeAttribute.Unlimited)>
        Public Property Xml As String
            Get
                Return xmlField
            End Get

            Set(ByVal value As String)
                SetPropertyValue("XML", xmlField, value)
            End Set
        End Property

        Public Property Name As String
            Get
                Return nameField
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Name", nameField, value)
            End Set
        End Property

        <Browsable(False)>
        Public Property OwnerId As String
            Get
                Return ownerIdField
            End Get

            Set(ByVal value As String)
                SetPropertyValue(Of String)("OwnerId", ownerIdField, value)
            End Set
        End Property

        Public Property IsShared As Boolean
            Get
                Return isSharedField
            End Get

            Set(ByVal value As Boolean)
                SetPropertyValue("IsShared", isSharedField, value)
            End Set
        End Property

        <Browsable(False)>
        Public Property ViewId As String
            Get
                Return viewIdField
            End Get

            Set(ByVal value As String)
                SetPropertyValue(Of String)("ViewId", viewIdField, value)
            End Set
        End Property

        Private Property ObjectSpace As IObjectSpace Implements IObjectSpaceLink.ObjectSpace
            Get
                Return objectSpaceField
            End Get

            Set(ByVal value As IObjectSpace)
                objectSpaceField = value
            End Set
        End Property
    End Class
End Namespace

File: CS/XPO/ViewSettings/ViewSettings.Module/BusinessObjects/SettingsStore.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp;
using System.ComponentModel;
using DevExpress.Xpo;

namespace ViewSettingsSolution.Module.BusinessObjects {
    public class SettingsStore : BaseObject, IObjectSpaceLink {
        private string xml;
        private string name;
        private string ownerId;
        private string viewId;
        private Boolean isShared;
        private IObjectSpace objectSpace;
        public SettingsStore(Session session)
            : base(session) {
        }
        [Browsable(false)]
        [Size(SizeAttribute.Unlimited)]
        public string Xml {
            get {
                return xml;
            }
            set {
                SetPropertyValue("XML", ref xml, value);
            }
        }
        public string Name {
            get {
                return name;
            }
            set {
                SetPropertyValue("Name", ref name, value);
            }
        }
        [Browsable(false)]
        public string OwnerId {
            get { return ownerId; }
            set { SetPropertyValue<string>("OwnerId", ref ownerId, value); }
        }
        public Boolean IsShared {
            get {
                return isShared;
            }
            set {
                SetPropertyValue("IsShared", ref isShared, value);
            }
        }
        [Browsable(false)]
        public string ViewId {
            get { return viewId; }
            set { SetPropertyValue<string>("ViewId", ref viewId, value); }
        }

        IObjectSpace IObjectSpaceLink.ObjectSpace {
            get {
                return objectSpace;
            }

            set {
                objectSpace = value;
            }
        }
    }
}
